Tests/Microsoft.Azure.Mobile.Test.Windows/Storage/FakeStorageTest.cs

```diff
@@ -56,19 +74,81 @@ public void ShutdownSucceed()
         public void GetLogsQueryError()
         {
             var mockAdapter = new Mock<IStorageAdapter>();
-            mockAdapter.Setup(adapter => adapter.OpenAsync()).Returns(TaskExtension.GetCompletedTask());
-            mockAdapter.Setup(adapter => adapter.ExecuteNonQueryAsync(It.IsAny<DbCommand>())).Returns(TaskExtension.GetCompletedTask());
-            mockAdapter.Setup(adapter => adapter.ExecuteQueryAsync(It.IsAny<DbCommand>())).Throws(new ConcreteDbException());
-
-            /* Return SqliteCommand because DbCommand is abstract */
-            mockAdapter.Setup(adapter => adapter.CreateCommand()).Returns(new SqliteCommand());
-
+            mockAdapter.Setup(
+                    a => a.GetAsync(It.IsAny<PredType>(), It.IsAny<int>()))
+                .Throws(new StorageException());
             var fakeStorage = new Mobile.Storage.Storage(mockAdapter.Object);
-
             var logs = new List<Log>();
             Assert.ThrowsException<StorageException>(() =>
                 fakeStorage.GetLogsAsync(StorageTestChannelName, 1, logs).RunNotAsync());
         }
 
+        /// <summary>
+        /// Verify that storage throws StorageException if something went wrong
+        /// </summary>
+        [TestMethod]
+        public void StorageThrowsStorageException()
+        {
+            var mockAdapter = new Mock<IStorageAdapter>();
+            mockAdapter.Setup(
+                    a => a.GetAsync(It.IsAny<PredType>(), It.IsAny<int>()))
+                .Throws(new StorageException());
+            mockAdapter.Setup(c => c.InsertAsync(It.IsAny<Mobile.Storage.Storage.LogEntry>()))
+                .Throws(new StorageException());
+            var fakeStorage = new Mobile.Storage.Storage(mockAdapter.Object);
+            try
+            {
+                fakeStorage.PutLogAsync("channel_name", new TestLog()).RunNotAsync();
+            }
+            catch (Exception e)
+            {
+                Assert.IsNotNull(e as StorageException);
+            }
```

Comment:
If fakeStorage.PutLogAsync().RunNotAsync() doesn't throw any exception, shouldn't this test fail? Can you modify your test to cover this scenario?